rama-http/src/layer/auth/require_authorization.rs

```diff
@@ -258,6 +260,53 @@ where
     }
 }
 
+pub(crate) struct AuthorizeContext<C> {
+    credential: C,
+    allow_anonymous: bool,
+}
+
+impl<C> AuthorizeContext<C> {
+    pub(crate) fn new(credential: C) -> Self {
+        Self {
+            credential,
+            allow_anonymous: false,
+        }
+    }
+}
+
+impl<S, B, C: ValidateRequest<S, B>> ValidateRequest<S, B> for AuthorizeContext<C>
```

Comment:
Why do you add this implementation? `C` is always a credential in this context. And we want to have control over what can be inside of `AuthorizeContext` that's why it's semi private. As such I don't see a reason for this impl block to exist at all?

Reply:
TBH I didn't consider updating the existing implementations for `Basic` and `Bearer` to include `AuthorizeContext`, I think because I was hesitant for the type to be more pervasive. I was a bit fixated on the "privateness" of `AuthorizeContext` which I see now was not necessary. Without adding it to the other `ValidateRequest` implementations I had no way (that I could see) of accessing the `allow_anonymous` value and so this was the approach I took. 
Given your suggestions I can totally see why that approach is nicer. Thank you for that.